microsetta_interface/templates/new_results_page.jinja2

```diff
@@ -1302,6 +1305,360 @@
+            .attr("height", height)
+            .append("g")
+            .attr("transform", `translate(${(width / 2) - 35}, ${height / 2})`);
+
+        const arc = d3.arc()
+            .innerRadius(radius * 0.57)
+            .outerRadius(radius * 0.82)
+            .startAngle(Math.PI * 0.75)
+            .endAngle(Math.PI * 2.15);
+
+        svg.append("path")
+            .attr("d", arc)
+            .attr("transform", `translate(-27.5, 15) rotate(100)`)            
+            .style("fill", "#D9E9EF")
+            .style("stroke", "#D9E9EF")
+            .style("stroke-width", 2);
+
+        svg.append("image")
+            .attr("xlink:href", "/static/img/Line_38.svg")
+            .style("transform", "rotate(-3.5deg)")
+            .attr("x", -radius - 11)
+            .attr("y", -radius + 213);
+
+        svg.append("image")
+            .attr("xlink:href", "/static/img/Line_38.svg")
+            .style("transform", "rotate(75.5deg)")
+            .attr("x", 75)
+            .attr("y", -radius + 85);
+
+        if(diversityInSampleValue == 2.40){
+            var scale = d3.scaleLinear()
+                .domain([2.40, 7.25])
+                .range([Math.PI * 0.75, Math.PI * 2.41]);
+
+            var angle = scale(diversityInSampleValue);
+
+            var ellipseXRadius = radius * 1.36;
+            var ellipseYRadius = radius * 1.13;
+
+            var x = Math.cos(angle) * ellipseXRadius;
+            var y = Math.sin(angle) * ellipseYRadius;
+
+            var tangentAngle = Math.atan2(
+                ellipseYRadius * Math.cos(angle),
+                -ellipseXRadius * Math.sin(angle)
+            );
+            var angleOffset = 3;
+            svg.append("image")
+                .attr("xlink:href", "/static/img/Line_40.svg")
+                .attr("x", x)
+                .attr("y", y)
+                .attr("transform", `rotate(${(tangentAngle * 180) / Math.PI + angleOffset}, ${x}, ${y})`);
+        }
+        
+        if(diversityInSampleValue == 7.25){
+            var scale = d3.scaleLinear()
+                .domain([2.40, 7.25])
+                .range([Math.PI * 0.76, Math.PI * 2.40]);
+
+            var angle = scale(diversityInSampleValue);
+
+            var ellipseXRadius = radius * 2.35;
+            var ellipseYRadius = radius * 0.72;
+
+            var x = Math.cos(angle) * ellipseXRadius;
+            var y = Math.sin(angle) * ellipseYRadius;
+
+            var tangentAngle = Math.atan2(
+                ellipseYRadius * Math.cos(angle),
+                -ellipseXRadius * Math.sin(angle)
+            );
+            var angleOffset = -54;
+            svg.append("image")
+                .attr("xlink:href", "/static/img/Line_40.svg")
+                .attr("x", x)
+                .attr("y", y)
+                .attr("transform", `rotate(${(tangentAngle * 180) / Math.PI + angleOffset}, ${x}, ${y})`);
+        }
+
+        if(diversityInSampleValue > 2.4 && diversityInSampleValue < 3.99){
```

Comment:
I'm not clear on why there's different logic for various possible ranges of diversity scores. Can it be consolidated into one calculation?

Reply:
I tried to and adjust the proportions accordingly but after a while I still couldn't keep the indicator line to stay nicely within the ellipse for all values.